lua/sniprun.lua

```diff
@@ -84,11 +94,26 @@ function M.setup(opts)
   M.config_up = 1
 end
 
+
+local highlight = function(group, styles)
+  local gui = styles.gui and 'gui='..styles.gui or 'gui=NONE'
+  local sp = styles.sp and 'guisp='..styles.sp or 'guisp=NONE'
+  local fg = styles.fg and 'guifg='..styles.fg or 'guifg=NONE'
+  local bg = styles.bg and 'guibg='..styles.bg or 'guibg=NONE'
+  -- print('highlighting '..group,gui,sp,fg,bg)
+  vim.api.nvim_command('autocmd ColorScheme * highlight '..group..' '..gui..' '..sp..' '..fg..' '..bg)
+end
+
+
 function M.setup_highlights()
-  vim.cmd("if !hlexists('SniprunVirtualTextOk')  \n hi SniprunVirtualTextOk	ctermbg=Cyan guibg=#66eeff ctermfg=Black guifg=#000000 \nendif")
-  vim.cmd("if !hlexists('SniprunVirtualTextErr') \n hi SniprunVirtualTextErr	ctermbg=DarkRed guibg=#881515 ctermfg=Black guifg=#000000 ")
-  vim.cmd("if !hlexists('SniprunFloatingWinErr') \n hi SniprunFloatingWinErr	guifg=#881515 ctermfg=DarkRed")
-  vim.cmd("if !hlexists('SniprunFloatingWinOk')  \n hi SniprunFloatingWinOk	ctermfg=Cyan guifg=#66eeff")
+  local colors_table = M.load_vimscript_config()["SniprunColors"]
+  vim.cmd('augroup snip_highlights')
+  vim.cmd('autocmd!')
```

Comment:
Is there a reason to use an autocmd instead of the highlight command ?

Reply:
yes. That scratched my brain as well. yesterday I was trying without the autocmds but without it, the highlight group clears out when the colorscheme loads in. ( Most or even all colorschemes has ```vim.cmd('hi clear')``` in it ) . So I thought of making a separate augroup . 

only ,If we could somehow append the highlight groups to $VIMRUNTIME/syntax/hitest.vim . without using autocmds .

Reply:
ok, it might just be my own (custom) colorscheme that doesn't do that. let's keep the autocommand then